arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/continuations/EagerEffectScope.kt

```diff
@@ -214,3 +214,17 @@ public suspend fun <R, B : Any> EagerEffectScope<R>.ensureNotNull(value: B?, shi
   contract { returns() implies (value != null) }
   return value ?: shift(shift())
 }
+
+/**
+ * Catch any possible `shift`ed outcome from [action], and transform it with [handler].
+ * This reads like a `catch` block, but for `EagerEffectScope`.
+ *
+ * The [handler] may `shift` into a different `EagerErrorScope`, which is useful to
+ * simulate re-throwing of exceptions.
+ */
+public fun<E, R, A> EagerEffectScope<E>.catch(
+  f: suspend EagerEffectScope<R>.() -> A,
+  recover: EagerEffectScope<E>.(R) -> A
+): A = eagerEffect(f).fold(
```

Comment:
NIT: Recover first, operation last.

Reply:
Also for a DSL like this? I know it's the convention for `bimap`, `fold` etc but to me it feels counter-intuitive here since this models an operation like `try { } catch { }` but for typed errors.

Reply:
Looking at things like [Flow's `catch`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/catch.html), I think that the convention is action, then recovery.